subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -115,13 +118,34 @@ int aws_fota_parse_DescribeJobExecution_rsp(const char *job_document,
 
 	cJSON *hostname = cJSON_GetObjectItemCaseSensitive(location, "host");
 	cJSON *path = cJSON_GetObjectItemCaseSensitive(location, "path");
+	cJSON *url = cJSON_GetObjectItemCaseSensitive(location, "url");
 
-	if ((cJSON_GetStringValue(hostname) != NULL)
+	if (cJSON_GetStringValue(url) != NULL) {
+		struct http_parser_url u;
+
+		http_parser_url_init(&u);
+		http_parser_parse_url(url->valuestring,
+				strlen(url->valuestring), false, &u);
+
+		uint16_t parsed_host_len =
```

Comment:
This block of code, until line:141 is not trivial to read. A comment or two would do good.